app/models/order_import.rb

```diff
@@ -131,8 +131,16 @@ def handle_save_nothing_on_error # TODO refactor and rename
   end
 
   def import_row(row) # TODO refactor
-    row_importer = OrderRowImporter.new(row, self)
-    row_importer.import
+    begin
+      row_importer = OrderRowImporter.new(row, self)
+      row_importer.import
+    rescue => e
+      ActiveRecord::Base.logger.error("Failed to bulk import: #{upload_file_path}")
+      ActiveRecord::Base.logger.error(e.backtrace)
+      ExceptionNotifier.notify_exception(e, data: { message: "Failed to bulk import: #{upload_file_path}" })
+      row_importer.add_error("Failed to import row")
+      set_error_mode
```

Comment:
It seems like there's enough going on in this `rescue` to make it extraction-worthy.